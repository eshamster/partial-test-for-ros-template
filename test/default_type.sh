#!/bin/bash

set -eu

root_dir="$(readlink -f $(dirname ${0}))/.."
. "${root_dir}/setenv"

test_name=test-default-type

function ros_t()
{
    ros template $@
}

ros_t deinit ${test_name} > /dev/null 2>&1 && :

ros_t init ${test_name}
ros_t checkout ${test_name}

(
    cd "${WORK_DIR}"

    echo test > file1
    ros_t add file1

    ros_t type

    echo "--- change default type ---"
    ros_t type djula
    echo test > file2
    ros_t add file2

    ros_t list
    ros_t type

    echo "--- add file1 again (doesn't change type) ---"
    ros_t add file1

    ros_t list
    ros_t type
)

ros_t deinit ${test_name}
