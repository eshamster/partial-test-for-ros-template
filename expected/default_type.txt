current default type is "copy"
--- change default type ---
      djula file2 
      copy  file1 
current default type is "djula"
--- add file1 again (doesn't change type) ---
      djula file2 
      copy  file1 
current default type is "djula"
